src/components/victory-axis.jsx

```diff
@@ -8,116 +8,251 @@ class VictoryAxis extends React.Component {
+      return "M" + sign * this.props.outerTickSize + "," + _.min(range) + "H0V" +
+        _.max(range) + "H" + sign * this.props.outerTickSize;
+    }
   }
 
-  getYScale() {
-    const scale = this.props.scale().range([this.state.yRange.max, this.state.yRange.min]);
-    return scale.domain([this.props.yDomain.min, this.props.yDomain.max]);
+  getActiveScale() {
+    const scale = this.state.scale;
+    if (scale.rangeBand) {
+      return (x) => scale(x) + scale.rangeBand() / 2;
+    } else {
+      return scale;
+    }
   }
 
+  getTickProperties() {
+    const orientation = this.props.orientation;
+    const verticalAxis = this.isVertical();
+    // determine axis orientation and layout
+    const sign = orientation === "top" || orientation === "left" ? -1 : 1;
+    // determine tick formatting constants based on orientationation and layout
+    const x = verticalAxis ? sign * this.state.tickSpacing : 0;
+    const y = verticalAxis ? 0 : sign * this.state.tickSpacing;
+    const x2 = verticalAxis ? sign * this.props.innerTickSize : 0;
+    const y2 = verticalAxis ? 0 : sign * this.props.innerTickSize;
+    let dy;
+    let textAnchor;
+    if (verticalAxis) {
+      dy = ".32em";
+      textAnchor = sign < 0 ? "end" : "start";
+    } else {
+      dy = sign < 0 ? "0em" : ".71em";
+      textAnchor = "middle";
+    }
+    return {x, y, x2, y2, dy, textAnchor};
+  }
 
-  componentDidMount() {
-    const xAxisFunction = d3.svg.axis()
-      .scale(this.getXScale())
-      .orient("bottom")
-      .ticks(this.state.ticks);
-    const yAxisFunction = d3.svg.axis()
-      .scale(this.getYScale())
-      .orient("left")
-      .ticks(this.state.ticks);
 
-    const xAxis = xAxisFunction(d3.select(React.findDOMNode(this.refs.xAxis)));
-    const yAxis = yAxisFunction(d3.select(React.findDOMNode(this.refs.yAxis)));
-    this.setState({xAxis, yAxis});
+  getTickLines() {
+    const verticalAxis = this.isVertical();
+    const ticks = this.getTicks();
+    const properties = this.getTickProperties();
+    let position;
+    let translate;
+    return _.map(ticks, (tick, index) => {
+      position = this.getActiveScale().call(this, tick);
+      translate = verticalAxis ?
+        "translate(0, " + position + ")" : "translate(" + position + ", 0)";
+      return (
+        <g key={"tick-" + index} className="tick" transform={translate}>
+          <line x2={properties.x2} y2={properties.y2} stroke={"black"} />
+          <text x={properties.x}
+            y={properties.y}
+            dy={properties.dy}
+            textAnchor={properties.textAnchor}
+            fill="black">
+            {this.getTickFormat().call(this, tick)}
+          </text>
+        </g>
+      );
+    });
+  }
+
+  getLabelElements() {
+    const style = this.getStyles();
+    const margin = this.getMarginValues();
+    const orientation = this.props.orientation;
+    const sign = orientation === "top" || orientation === "left" ? -1 : 1;
+    if (this.isVertical()) {
```

Comment:
you can put ternaries inside of the individual prop curly brackets, so `transform={this.isVertical() ? "rotate(-90)" : ""}` would work instead of wrapping the whole return in conditionals